### PR TITLE
feat(discover-snql): Add support for tags

### PR DESCRIPTION
**Pull request by Zylphrex**
This adds support for tags in snql.

### CODE REVIEW

File: src/sentry/search/events/base.py [hunk 1]
```diff
@@ -39,5 +39,16 @@ def project_slugs(self) -> Mapping[str, int]:
 
         return {p.slug: p.id for p in project_slugs}
 
-    def column(self, name: str) -> Column:
-        return Column(self.resolve_column_name(name))
+    def column(self, name: str, alias: Optional[str] = None) -> Column:
```
Comment (wmak) [inline src/sentry/search/events/base.py:42]: Thoughts on instead introducing a new `class AliasedColumn` to match what snql will eventually add?
since AliasedColumns won't be usable in Functions, we can preemptively handle that as we continue building with it?

File: src/sentry/search/events/base.py [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
 
 from django.utils.functional import cached_property
 from snuba_sdk.column import Column
-from snuba_sdk.function import CurriedFunction
+from snuba_sdk.function import CurriedFunction, Function
 from snuba_sdk.orderby import OrderBy
 
 from sentry.models import Project
-from sentry.search.events.constants import SNQL_FIELD_ALLOWLIST
+from sentry.search.events.constants import SNQL_FIELD_ALLOWLIST, TAG_KEY_RE
```
Comment (wmak) [inline src/sentry/search/events/base.py:9]: We can do in another PR if you want, but I think with tag support its time we just drop the allowlist altogether now, the only things that aren't "allowed" now are unimplemented aliases&functions I think

File: tests/sentry/snuba/test_discover.py [hunk 3]
```diff
@@ -238,6 +239,97 @@ def test_issue_filters(self):
...
+            ("key1", "has:key1", ["value1", "value2"]),
+            ("key1", "!has:key1", []),
+            ("key1", "key1:value1", ["value1"]),
+            ("key1", "key1:value2", ["value2"]),
+            ("key1", 'key1:""', []),
+            ("key1", "key1:value*", ["value1", "value2"]),
+            ("key1", 'key1:["value1"]', ["value1"]),
+            ("key1", 'key1:["value1", "value2"]', ["value1", "value2"]),
+            ("tags[key1]", "", ["value1", "value2"]),
+            # has does not work with tags[...] syntax
```
Comment (wmak) [inline tests/sentry/snuba/test_discover.py:305]: 😞 TIL

File: src/sentry/search/events/base.py (hunk 1 above)
Comment (Zylphrex) [reply src/sentry/search/events/base.py:42]: Ahh yes, just did this to get things working for now, will introduce a `AliasedColumn` soon 👍

Comment (Zylphrex) [reply src/sentry/search/events/base.py:40]: Instead of a class, i opted for just a function `aliased_column` that returns a `SelectType`. A class would've been just more boilerplate

File: src/sentry/search/events/base.py (hunk 2 above)
Comment (Zylphrex) [reply src/sentry/search/events/base.py]: Removed

File: src/sentry/search/events/base.py (hunk 1 above)
Comment (wmak) [reply src/sentry/search/events/base.py:40]: Actually could we update this so it doesn't accept the alias? Reason being that Function's likely won't accept aliased_columns as parameters and we'll want the plain resolved column in those cases.
(also add something along those lines to the docstring)

Comment (Zylphrex) [reply src/sentry/search/events/base.py:40]: Not sure what you mean by this. If it doesn't accept `alias`, how would it know that the column needs to be aliased? The `alias` arg is intentionally optional to handle the use case with functions where it doesn't need and may not work with the alias.

Comment (Zylphrex) [reply src/sentry/search/events/base.py]: resolved in 247dbfc

**Review by wmak — APPROVED** [review]
